packages/x-data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -615,6 +618,70 @@ describe('<DataGrid /> - Pagination', () => {
     });
   });
 
+  describe('resetPageOnSortFilter prop', () => {
+    it('should reset page to 0 if sort or filter is applied and `resetPageOnSortFilter` is `true`', () => {
+      const { setProps } = render(
+        <BaselineTestCase
+          initialState={{ pagination: { paginationModel: { page: 0, pageSize: 5 }, rowCount: 0 } }}
+          pageSizeOptions={[5]}
+        />,
+      );
+
+      act(() => {
+        apiRef.current?.setPage(1);
+      });
+      expect(apiRef.current!.state.pagination.paginationModel.page).to.equal(1);
+
+      act(() => {
+        apiRef.current?.sortColumn('id', 'desc');
```

Comment:
Use `!` throughout instead of `?`?
```suggestion
        apiRef.current!.sortColumn('id', 'desc');
```

Reply:
I plan to do this for the whole repo in another PR